docs/Instance-Variable-Assumption.md

```diff
@@ -64,6 +64,50 @@ would report:
   [1]:InstanceVariableAssumption: Dummy assumes too much for instance variable @ivar [https://github.com/troessner/reek/blob/master/docs/Instance-Variable-Assumption.md]
 ```
 
+Note that this example would trigger this smell warning as well:
+
+```Ruby
+class Parent
+  def initialize(omg)
+    @omg = omg
+  end
+end
+
+class Child < Parent
+  def initialize(omg)
+    super
+  end
+
+  def foo
+    @omg
+  end
+end
+```
+
+The way to address the smell warning is that you should create an `attr_reader` to use `@omg` in the subclass and not access `@omg` directly like this:
+
+```Ruby
+class Parent
+  attr_reader :omg
+
+  def initialize(omg)
+    @omg = omg
+  end
+end
+
+class Child < Parent
+  def initialize(omg)
+    super
+  end
+
+  def foo
+    omg
+  end
+end
+```
+
+Directly accessing instance variables is considered a smell because it [breaks encapsulation](http://designisrefactoring.com/2015/03/29/organizing-data-self-encapsulation/) and makes it harder to reason about code.
```

Comment:
>> That article goes a lot further than our smell detectors.

I have to disagree on this. It explains why to prefer setters and getter over direct access which is what we should explain here.

>> Or maybe we can explain it ourselves?

What do you feel is missing in the article?